### PR TITLE
Allow matcher to accept Axe results object

### PR DESCRIPTION
**Pull request by gabalafou**
## Current behavior before this pull request

You cannot pass anything other than a Playwright handle (page, locator, etc.) to the Expect.js matcher:

```js
expect(page).toBeAccessible()
```

## Behavior after this pull request

After this pull request, you should be able to pass either a Playwright handle or an [Axe results object](https://www.deque.com/axe/core-documentation/api-documentation/#results-object):

```js
const { results } = await waitForAxeResults(page);
expect(results).toBeAccessible();
```

## Background / motivation

Refer to discussion in [issue #21](https://github.com/Widen/expect-axe-playwright/issues/21)

### CODE REVIEW

File: README.md
Comment (gabalafou) [inline README.md]: If we want the matcher to support an Axe results object and for the resulting line of code to read in a semantically coherent and accurate way, it may make sense to [change the matcher name](https://github.com/Widen/expect-axe-playwright/pull/24). That way this would change to the more sensical:

```js
expect(results).toHaveNoAxeViolations()
```

File: src/utils/axe.ts
Comment (gabalafou) [inline src/utils/axe.ts]: Couldn't think of a better name, sorry.

**Review by gabalafou — COMMENTED** [review]
done self-reviewing

File: src/matchers/toBeAccessible/index.ts
Comment (mskelton) [inline src/matchers/toBeAccessible/index.ts]: Removing this is incorrect as it breaks the ability to define the filename at the project level.

File: src/axe/index.ts
Comment (mskelton) [inline src/axe/index.ts]: The polling behavior feels like something that should be part of the matcher, not this low level utility.

File: src/axe/index.ts
Comment (mskelton) [inline src/axe/index.ts]: This should be defined inside of `utils/axe` rather than the index file.

File: src/utils/axe.ts
Comment (mskelton) [inline src/utils/axe.ts]: `checkLocator` perhaps?

File: src/matchers/toBeAccessible/index.ts
Comment (mskelton) [inline src/matchers/toBeAccessible/index.ts]: This code feels a bit messy.  A helper function that does the conditional and returns separate objects would be cleaner.

**Review by gabalafou — COMMENTED** [review]
Take 2! :)
I tried to implement all review suggestions.

File: src/matchers/toBeAccessible/index.ts
Comment (mskelton) [inline src/matchers/toBeAccessible/index.ts]:
```suggestion
      return waitForAxeResults(obj as Handle, options)
```

File: src/waitForAxeResults.spec.ts
Comment (mskelton) [inline src/waitForAxeResults.spec.ts]: Always prefer `toBe(true)` as it tests that an actual boolean was returned.

```suggestion
    const { ok } = await waitForAxeResults(page)
    await expect(ok).toBe(true)
```

Comment (mskelton) [reply src/waitForAxeResults.spec.ts]: Additionally, could we test the results object that is returned?

File: src/waitForAxeResults.spec.ts
Comment (mskelton) [inline src/waitForAxeResults.spec.ts]: Same about testing the results object and using `toBe(false)`

File: src/waitForAxeResults.ts
Comment (mskelton) [inline src/waitForAxeResults.ts]: Unnecessary default export since you are already exporting as a named export

```suggestion
```

**Comment by gabalafou** [comment]
Thanks for the code review! How does it look now?

**Comment by mskelton** [comment]
Looks like there are some ESLint and Prettier issues.  Once those are fixed, looks good.

**Comment by mskelton** [comment]
Forgot to mention, this needs a changeset with a version bump (minor) and change notes.

File: CHANGELOG.md
Comment (gabalafou) [inline CHANGELOG.md]: Hmm. Should I squash the commits and force push on this PR before referencing the commit SHA here? 

I added this commit SHA manually, but it refers to only one of several commits on this PR branch.

Comment (mskelton) [reply CHANGELOG.md]: The changelog should not be manually edited.  Please add a changeset with the change notes.

https://github.com/changesets/changesets/blob/main/docs/adding-a-changeset.md#i-am-in-a-single-package-repository

Comment (gabalafou) [reply CHANGELOG.md]: Okay, second attempt :) 

I read those instructions yesterday, but I've never used this tool before, so when I ran `yarn changeset` yesterday and it added a file to the `.changesets/` directory, and I saw that there were no other files in that directory, I thought that meant that I did something wrong or that there were more steps I needed to complete. So then I ran `yarn changeset version`, which added the note to the change log and the version bump. 

To satisfy my own curiosity, I'm trying to wrap my mind around how this works. If you have the time, could you read what I've written below and see if I'm on the right track? No worries if not :)

Looking at the git log of the `.changesets` directory, it looks to me that something like the following happens:

1. Create a PR, push commits
2. Add a changeset file to the `.changesets` directory. (Via `yarn changeset` and committing and pushing the result).
3. Squash-merge the commits on the PR. (I assume this is how Changeset knows which SHA to use in the change log. Although I suppose it's probably smart enough to use the SHA of a merge commit.)
4. The GitHub action you've configured (on push to `main`) uses the changesets/action to create a pull request that is essentially the output of `yarn changeset version`.
5. When you merge that PR into main, the changesets/action gets triggered a second time, but this time it runs the `yarn release` command? 

Out of curiosity, how do the GitHub releases get added and/or how are the commits tagged with the version number on GitHub? I'm guessing the Changesets Release Action takes care of all that? Since it has an input called `createGithubReleases`, defined as:

- createGithubReleases - A boolean value to indicate whether to create Github releases after publish or not. Default to true

File: src/matchers/toBeAccessible/index.ts
Comment (mskelton) [inline src/matchers/toBeAccessible/index.ts]: This merge needs to happen before calling `getResults` and the `opts` object needs to be passed to `getResults`.  This allows for defining config at the project level.

File: src/waitForAxeResults.ts
Comment (mskelton) [inline src/waitForAxeResults.ts]: Probably don't need the explicit return type, TypeScript should be able to infer it.

```suggestion
) {
```

File: src/matchers/toBeAccessible/index.ts
Comment (mskelton) [reply src/matchers/toBeAccessible/index.ts]: Oh, looks like `getResults` is doing that already.  Interesting

**Review by mskelton — APPROVED** [review]

File: CHANGELOG.md
Comment (mskelton) [reply CHANGELOG.md]: Yep, you're right on track with how changesets works.  And yes, the action creates the GitHub release as well.  It's honestly such an awesome tool for versioning, way superior to the old Lerna flow we used to use for our monorepos before.  The other awesome part is it's all PR based, so you can have strict controls on push access, everything including versioning goes through a pull request, so no one needs direct push access when updating versions and publishing.

Comment (gabalafou) [reply CHANGELOG.md]: It looks really neat!

Okay so if I may bother you with another question... 😁 

One thing I wonder, since the changeset file only records the delta of the semantic versioning, meaning it only records whether the set of changes represent major, minor, or patch version upgrades, it seems that the order in which PRs get merged matters.

Scenario 1:

1. PR #23 is merged, package goes from version 2.2.1 to 2.3.0
3. PR #24 is merged, version 2.3.0 to 3.0.0

Scenario 2:

1. PR #24 is merged, package goes from version 2.2.1 to 3.0.0
3. PR #23 is merged, version 3.0.0 to 3.0.1

In my particular case with these two PRs, I think either order is fine, but it seems like a pretty heavy assumption that this is generally the case, especially where major changes are concerned.

Comment (mskelton) [reply CHANGELOG.md]: Order doesn't matter, all changesets will be merged to create the final version using the highest version bump.  This allows notes in the changelog to still be separated by major/minor/patch, but the version bump will be correct.  In this case, regardless of the merge order the new version will be 3.0.0.

Of course, that's assuming I wait to release until this PR is merged, which I'm planning on doing 🙂

Comment (gabalafou) [reply CHANGELOG.md]: That makes sense. Neat!

Thank you for answering my questions. 😃

File: src/matchers/toBeAccessible/index.ts
Comment (gabalafou) [reply src/matchers/toBeAccessible/index.ts]: Yeah, there's some repetition, but I'm not sure how to avoid it. 

Do you want me to try to refactor this in a way that avoids the repetition? Or do you want some time to think about it?

File: src/waitForAxeResults.ts
Comment (gabalafou) [inline src/waitForAxeResults.ts]: When @mskelton [removed the return type annotation](https://github.com/Widen/expect-axe-playwright/pull/23/commits/1362f15034ef481d0ef01a296c6cff01e6db635a) from `waitForAxeResults`, this import became obsolete.

```suggestion
import type { RunOptions } from 'axe-core'
```

**Comment by mskelton** [comment]
Looks like there are merge conflicts

File: src/matchers/toBeAccessible/index.ts
Comment (mskelton) [reply src/matchers/toBeAccessible/index.ts]: Nah, it's fine.  You could make a function that does it so the `test.info` and `merge` calls are consistent if you want.

File: src/utils/options.ts
Comment (mskelton) [inline src/utils/options.ts]:
```suggestion
  return merge(info.project.use.axeOptions, options)
```

**Review by mskelton — APPROVED** [review]
Great work!

**Pull request merged**
